src/main/scala/zio/ftp/UnsecureFtp.scala

```diff
@@ -39,7 +43,14 @@ final private class UnsecureFtp(unsafeClient: Client) extends FtpAccessors[Clien
       .fail(
         FileTransferIncompleteError(s"Cannot finalize the file transfer and completely read the entire file $path.")
       )
-      .unlessM(execute(_.completePendingCommand()))
+      .unlessM {
+        val complete = execute(_.completePendingCommand())
+        dataTimeout match {
+          case None          => complete // if the server never responds to the complete, this will freeze forever
+          case Some(timeout) => complete.timeout(timeout).someOrElse(false)
+        }
+      }
+      .provideSomeLayer[Blocking](env)
```

Comment:
Sorry for my late response. :(

can we change the signature of the method `ZStream[Blocking & Clock, IOException, Byte]`